packages/api-client-core/src/ModelManager.ts

```diff
@@ -8,5 +8,8 @@ import { GadgetRecordList } from "./GadgetRecordList";
 export interface AnyModelManager {
   connection: GadgetConnection;
   findOne(id: string, options: any): Promise<GadgetRecord<any>>;
+  maybeFindOne(id: string, options: any): Promise<GadgetRecord<any> | null>;
   findMany(options: any): Promise<GadgetRecordList<any>>;
+  findFirst(options: any): Promise<GadgetRecord<any>>;
+  maybeFindFirst(options: any): Promise<GadgetRecord<any> | null>;
```

Comment:
Maybe I'm misunderstanding what you're saying, but the behaviour Harry and I discussed was:
- `find__` methods return data on success, and error when no data is found (this is currently already true).
- `maybeFind__` methods return data on success, and null when no data is found; they only error on other issues.

Reply:
I think you're right @thegedge that the maybeXYZ functions are rather light syntatic sugar, but, I think the needs for them are so frequent that they merit inclusion. In Gadget's code (with `typeorm`) at least we use `db.findOne` and `db.findOneOrFail` both a lot:

```
inspector ~/C/gadget (use-data-model-query-race*) ➜  ag "db.findOne\(" -c | wc -l
23
inspector ~/C/gadget (use-data-model-query-race*) ➜  ag "db.findOneOrFail\(" -c | wc -l
24
```

It is true that you could just wrap one to make the other, but, I think its convenient to not have to write that wrapper all the time, you know? FWIW ActiveRecord has the same thing with `.find` and `.find!`, but notably Django does not because of the pythonic explicit-is-better-than-implicit thing, see https://stackoverflow.com/questions/3090302/how-do-i-get-the-object-if-it-exists-or-none-if-it-does-not-exist-in-django . I have a preference towards this convenience as opposed to explicitness because it allows us as the platform to do more for the user, instead of requiring them to think about how to write that wrapper all the time.

Reply:
I guess what I'm saying is we're adding `maybeFind(x)` which appears to just be `find(x).catch(() => null)`. It's low effort for people to do this today, so I'm hesitant to add because a larger interface hinders discoverability (and less of an issue, but increases bundle size).

I guess, more generally, my preference would be that that we don't add things that are easy to accomplish with the current thing unless we have feedback indicating it's worth adding (maybe we have?).

Reply:
FWIW I'm not opposed, just hesitant.

If we do decide to introduce these, it seems reasonable to build one on top of the other though, to avoid diverging implementations (and I don't think we'd ever want these to diverge?)